### PR TITLE
WIP: feat: precommit hooks added

### PR DESCRIPTION
**Pull request by savannahar68**
TODO: Run `pre-commit run --all-files` and resolve errors

### CODE REVIEW

**Comment by vinaykakade** [comment]
Optionally, let us add the "ignored" tests as well as "loom" tests.

* Ignored in infino -
`cargo test -- --ignored`

* loom in tsldb -
`RUSTFLAGS="--cfg loom" cargo test --test loom --release`

**Comment by savannahar68** [comment]
@vinaykakade commented fmt

It's throwing error when run via pre commit hook with this error 

```
fmt......................................................................Failed
- hook id: fmt
- exit code: 1

Error writing files: failed to resolve mod `tsldb`: /Users/savannahar/Desktop/code/tsldb/infino/tsldb/tsldb.rs does not exist
```

Looks like some issue with workspace setup. Need your help here.

**Comment by vinaykakade** [comment]
Pushed a minor PR #18, after that `cargo fmt --all -- --check` is working for me.

In the command output below, the circular directory `tsldb/infino/tsldb` is confusing. Ideally, should just be `infino/tsldb`. 

> @vinaykakade commented fmt
> 
> It's throwing error when run via pre commit hook with this error
> 
> ```
> fmt......................................................................Failed
> - hook id: fmt
> - exit code: 1
> 
> Error writing files: failed to resolve mod `tsldb`: /Users/savannahar/Desktop/code/tsldb/infino/tsldb/tsldb.rs does not exist
> ```
> 
> Looks like some issue with workspace setup. Need your help here.

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+      - id: cargo-test-infino-ignored
+        name: cargo test infino ignored
+        description: Run Ignored cargo test
+        entry: cargo test -- --ignored
+        language: system
+        types: [rust]
+        pass_filenames: false
+      - id: cargo-test-tsldb-loom
+        name: cargo test infino loop
+        description: Run loop cargo test
```
Comment (vinaykakade) [inline .pre-commit-config.yaml:63]: nit: 'loop cargo test' -> 'loom cargo test'

**Comment by savannahar68** [comment]
@vinaykakade fixed the issue good to merge

**Pull request merged**